src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1665,16 +1642,27 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             }
         }
 
+        if (empty($config['transports'])) {
+            $container->removeDefinition('messenger.transport.symfony_serializer');
+            $container->removeDefinition('messenger.transport.amqp.factory');
+        } else {
+            $container->getDefinition('messenger.transport.symfony_serializer')
+                ->replaceArgument(1, $config['serializer']['format'])
+                ->replaceArgument(2, $config['serializer']['context']);
+            $container->setAlias('messenger.transport.serializer', $config['serializer']['id']);
```

Comment:
Now that it is only the default serializer, shouldn't we rename it `messenger.default_serializer`? (similar to `messenger.default_bus`)

Reply:
Same for the config option IMHO.
Also a bit unrelated, but now that the default serializer is the native one, the `context` and `format` options may be a bit confusing at first read. Should we add an `->info` to indicates it's only used by the `messenger.transport.symfony_serializer`?

Reply:
Good catches all around!

1) Changed the service to `messenger.default_serializer`
2) In the config, I actually split the config into two pieces, based on @ogizanagi's feedback:

```yml
# before
framework:
    messenger:
        serializer:
            id: messenger.transport.native_php_serializer
            format: json
```

```yml
# after
framework:
    messenger:
        default_serializer: messenger.transport.native_php_serializer
        symfony_serializer:
            format: json
```